### PR TITLE
Allow for non parties to have auto loans.

### PR DESCRIPTION
**Pull request by darko-da**
Add logic to deal with a Non-Party being the holder of the account.

Additional cleanup of language and naming.

### CODE REVIEW

File: daml-model/daml/Model/LoanConfig.daml [hunk 1]
```diff
@@ -36,18 +39,20 @@ template LoanConfig
 
     where
         signatory provider
-        observer accountHolder
+        observer getObserver accountHolder
```
Comment (darko-da) [inline daml-model/daml/Model/LoanConfig.daml:42]: Is there a neater way of doing this?

File: daml-model/daml/Model/LoanConfig.daml [hunk 2]
```diff
@@ -36,18 +39,20 @@ template LoanConfig
...
         key (BalanceKey with iban = account, provider = provider) : BalanceKey
         maintainer key.provider
 
         ensure activeLoan >= 0.0
             && activeLoan <= maxLoanSize
             && balanceLoanThreshold >= 0.0
-            && balanceLoanThreshold <= maxLoanSize
```
Comment (darko-da) [inline daml-model/daml/Model/LoanConfig.daml:46]: maxLoanSize is independent of the threshold.
i.e. you could say "I'm willing to give you a loan up to 1000 USD, and you can top up your account for any value below 1'000'000 .. i.e. if you get to 999'500, you can get a 500USD loan to bump it up to a million again."

File: daml-model/daml/Model/LoanConfig.daml [hunk 3]
```diff
@@ -36,18 +39,21 @@ template LoanConfig
...
-            && balancePaybackThreshold > maxLoanSize
+            && balancePaybackThreshold >= balanceLoanThreshold
```
Comment (darko-da) [inline daml-model/daml/Model/LoanConfig.daml:49]: This is to make sure we don't get into a loan taking <-> payback loop by having the threshold of the payback being lower than the loan taking.

File: daml-model/daml/Model/LoanConfig.daml (hunk 1 above)
Comment (darko-da) [reply daml-model/daml/Model/LoanConfig.daml]: I found a neater way :)

File: daml-model/daml/Model/LoanConfig.daml [hunk 4]
```diff
@@ -36,18 +39,17 @@ template LoanConfig
 
     where
         signatory provider
-        observer accountHolder
+        observer optionalToList accountHolder
```
Comment (tamaskalcza-da) [inline daml-model/daml/Model/LoanConfig.daml:42]: No need (see: https://docs.daml.com/daml/stdlib/Prelude.html#class-da-internal-template-functions-isparties-53750):
```suggestion
        observer accountHolder
```

File: daml-model/daml/Model/LoanConfig.daml (hunk 3 above)
Comment (mjuchli-da) [reply daml-model/daml/Model/LoanConfig.daml:49]: Makes sense

File: daml-model/daml/Model/LoanConfig.daml (hunk 2 above)
Comment (mjuchli-da) [reply daml-model/daml/Model/LoanConfig.daml:46]: Agreed

**Review by tamaskalcza-da — APPROVED** [review]

**Review by mjuchli-da — APPROVED** [review]

**Pull request merged**